datafusion/core/tests/provider_filter_pushdown.rs

```diff
@@ -146,7 +147,20 @@ impl TableProvider for CustomProvider {
         match &filters[0] {
             Expr::BinaryExpr { right, .. } => {
                 let int_value = match &**right {
+                    Expr::Literal(ScalarValue::Int8(i)) => i.unwrap() as i64,
```

Comment:
I think you might want do avoid doing this for NULLs (aka `None`) values

Something like:

```suggestion
                    Expr::Literal(ScalarValue::Int8(Some(i))) => i as i64,
```

Reply:
thanks for you comments, I just follow the original implementation.
But I will follow your nice comments.